### ISSUE TITLE
Starting to quote a message that then expires before/while submitting gives unexpected behaviour

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/spreed#7994]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Set up expiration
2. Post a message
3. Click on reply of **2**
4. Wait for the message to expire
5. Try to submit the form

---

Also tagging api, as there is a chance that the message expires exactly while you post, which will result in a "default" `400 Bad Request`.
Maybe we should add a message there on the return which can then be shown in the UIs

**Comment by nickvergessen** [comment]
![Bildschirmfoto vom 2022-09-23 07-47-34](https://user-images.githubusercontent.com/213943/191898918-09217c9f-4d39-4147-8abd-fd4c828c4866.png)

**Comment by vitormattos** [comment]
Yes, only will be deleted by cron job, but when I reproduced the scenario I didn't saw the deleted message, this is the point.

If I don't see and can't retrieve the deleted message, I think that wait by cron job won't a problem.

Could you provide more information about how to reproduce the scenario that you got when took the print screen? I think that this problem already was solved.

**Comment by vitormattos** [comment]
I think that this PR https://github.com/nextcloud/spreed/pull/8515 solved the problem.

**Comment by vitormattos** [comment]
This scenario is different from this issue description. Isn't about expire feature, is related as deleted history.

I updated the subject and description.

**Comment by vitormattos** [comment]
Suggested response:

**Status code**: `400`
**Body**:
```json
{
    "type": "error",
    "error": {
        "code": "invalid_parent_message",
        "message": "The parent message do not exists anymore."
    }
}
```

But, we need the opinion from client developers.

**Comment by nickvergessen** [comment]
Could also just drop the "reply"/parent reference and post it as a normal message.
In both cases (expiration and purge of history) it's more to what I would expect as a user.

**Comment by vitormattos** [comment]
Maybe the message will sent outside of context. I think that block to send would be best but this is more a UX point.

**Comment by vitormattos** [comment]
@nickvergessen is ok the acceptance scenario to this issue?

**Comment by nickvergessen** [comment]
Yeah I guess blocking would also work, but yeah a dedicated error should be returned.

**Comment by vitormattos** [comment]
Then, I think the scenario at description of this issue is OK to be implemented because have a specific error message.

I think that will be necessary check with clients developers what will be the impact of this change before start the implementation.

**Comment by Antreesy** [comment]
> Could also just drop the "reply"/parent reference and post it as a normal message.

Can be done for frontend before posting while processing messages (or purging the messages store) with dropping `parentIdToReply` in chatExtras cc @DorraJaouad

**Comment by nickvergessen** [comment]
Could still happen that it expires within the network time 🙈

**Comment by DorraJaouad** [comment]
> Could still happen that it expires within the network time

Even if it is expires ( no sys message to process), the reply parent in Quote no longer has id so `parentIdToReply` can be dropped based on that instead of processing messages.

**Comment by Antreesy** [comment]
> before posting

We could only fix it at frontend untill request was sent. Rest is on API)
